wat

```diff
@@ -1,6 +1,10 @@
 #!/bin/sh
 log_lines() {
-  local regex='^.*\(starting full system upgrade\|upgraded.*\)$'
+  local installed=''
+  if [ "x$SHOW_INSTALLED" == "x1" ]; then
```

Comment:
That would work, but I didn't want to restrict future development: If you make use of positional arguments, then you've got to pass every single option, only to then declare a local variable to describe each positional argument. (Which you then need to check w/ an if..fi block anyways)

By using global variables, the data maintains meaning from initial definition (inheriting env from user and/or from command line arguments) through to where it's ultimately used. Ideally, if this were python (or some other dict() capable lang) we could just pass a "bag of options" to avoid cluttering the global namespace.

Ultimately: It's your project. If I can't sway your opinion, I'll alter the code to match your request. ;-)